presentation/src/main/java/com/knocklock/presentation/lockscreen/mapper/NotificationMapper.kt

```diff
@@ -0,0 +1,157 @@
+ *
+ * StatusBarNotification에서 title이나 content가 있는 경우 데이터베아스의 저장할 키를 생성합니다.
+ * @param packageManager AppTitle을 가져오기 위한 packageManager입니다.
+ * @return null OR Group.Key
+ */
+fun StatusBarNotification.getDatabaseKey(packageManager: PackageManager): String? {
+    val title: String = convertString(this.notification.extras.getCharSequence("android.title"))
+    val content: String = convertString(this.notification.extras.getCharSequence("android.text"))
+    return if (title != "" || content != "") {
+        val appTitle = getDrawableAndAppTitle(packageManager, packageName).first ?: ""
+        val subText: String = convertString(this.notification.extras.getCharSequence("android.subText"))
+
+        if (subText == "")getGroupKey(packageName, appTitle, title) else getGroupKey(packageName, appTitle, subText)
+    } else {
+        null
+    }
+}
+
+/**
+ * StatusBarNotification을 Domain Layer Model로 변환합니다.
+ *
+ * StatusBarNotification에서 title이나 content가 있는 경우 Notification Model(Domain)을 생성합니다.
+ *
+ * @param packageManager AppTitle을 가져오기 위한 packageManager입니다.
+ * @return null OR Notification Domain Model
+ */
+fun StatusBarNotification.toModel(packageManager: PackageManager): NotificationModel? {
+    val title: String = convertString(this.notification.extras.getCharSequence("android.title"))
+    val content: String = convertString(this.notification.extras.getCharSequence("android.text"))
+    return if (title != "" || content != "") {
+        val subText: String = convertString(this.notification.extras.getCharSequence("android.subText"))
+        val packageName = packageName
+
+        val intent: PendingIntent? = notification.contentIntent
+
+        val appTitle = getDrawableAndAppTitle(packageManager, packageName).first ?: ""
+
+        NotificationModel(
+            id = key,
+            packageName = packageName,
+            postedTime = postTime,
+            appTitle = if (subText == "") appTitle else subText,
+            title = title,
+            content = content,
+            isClearable = isClearable,
+            groupKey = if (subText == "") getGroupKey(packageName, appTitle, title) else getGroupKey(packageName, appTitle, subText)
+        )
+    } else {
+        null
+    }
+}
+
+fun getGroupKey(packageName: String, appTitle: String, title: String): String {
+    return packageName + appTitle + title
+}
+
+fun convertString(var1: Any?): String {
+    return var1?.toString() ?: ""
+}
+
+/**
+ * Notification Model(Domain)을 Presentation Layer Model로 변환합니다.
+ *
+ * packageName과 packageManager를 통해 AppTitle과 AppIcon Drawable을 얻습니다.
+ * * * Notification Model(Domain)에 저장된 appTitle(SubText)가 없는 경우, [getDrawableAndAppTitle]을 통해 얻은 AppTitle로 지정합니다.
+ * @param packageManager AppTitle을 가져오기 위한 packageManager입니다.
+ * @return [Notification]
+ */
+fun NotificationModel.toModel(packageManager: PackageManager): Notification {
+    val temp = getDrawableAndAppTitle(packageManager, packageName)
```

Comment:
```suggestion
    val (title, icon) = getDrawableAndAppTitle(packageManager, packageName)
```
Pair를 위처럼 받을 수 있어요. 가독성 🆙

Reply:
@jaeryo2357 [b4f68e1](https://github.com/Knock-Lock/Knock-Lock/pull/101/commits/b4f68e10bbc60762f74fede162e832aaa76f6b33) 수정했습니다